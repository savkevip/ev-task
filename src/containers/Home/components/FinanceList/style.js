import styled from "styled-components";

export const FinanceListWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const ItemList = styled.ul`
  width: 100%;
  max-height: 400px;
  overflow-x: hidden;
`;

export const Item = styled.li`
  display: flex;
  justify-content: space-between;
  text-align: center;
`;
