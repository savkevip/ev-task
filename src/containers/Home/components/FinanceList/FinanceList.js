import React from "react";
import PropTypes from "prop-types";
import { FinanceListWrapper, ItemList, Item } from "./style";

const FinanceList = ({ list }) => (
  <FinanceListWrapper>
    <ItemList className="list-group">
      <Item className="list-group-item active">
        <span className="col-xs-4">From</span>
        <span className="col-xs-4">Description</span>
        <span className="col-xs-4">Amount</span>
        <span className="col-xs-4">Date</span>
      </Item>
      {list.map((item, i) => (
        <Item key={`${item.description}${i}`} className="list-group-item">
          <span className="col-xs-4">{item.from || 'Milos'}</span>
          <span className="col-xs-4">{item.description}</span>
          <span className="col-xs-4">{item.amount}</span>
          <span className="col-xs-4">{new Date(item.date).getFullYear()}</span>
        </Item>
      ))}
    </ItemList>
  </FinanceListWrapper>
);

FinanceList.propTypes = {
  list: PropTypes.array
};

export default FinanceList;
