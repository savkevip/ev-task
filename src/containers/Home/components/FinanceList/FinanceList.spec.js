import React from "react";
import { shallow } from "enzyme";
import FinanceList from "./FinanceList";

const list = [
  {
    amount: 10.5,
    date: "2017-02-10T09:22:20.000Z",
    description: "Pizza",
    from: "Mendy"
  },
  {
    amount: 10.5,
    date: "2016-01-10T09:20:00.000Z",
    description: "Diner",
    from: "Wendy"
  }
];

describe("FinanceList", () => {
  it("should match snapshot", function() {
    const component = shallow(<FinanceList list={list} />);
    expect(component).toMatchSnapshot();
  });
});
