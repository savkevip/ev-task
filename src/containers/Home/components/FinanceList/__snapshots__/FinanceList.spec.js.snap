// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinanceList should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FinanceList
    list={
        Array [
            Object {
              "amount": 10.5,
              "date": "2017-02-10T09:22:20.000Z",
              "description": "Pizza",
              "from": "Mendy",
            },
            Object {
              "amount": 10.5,
              "date": "2016-01-10T09:20:00.000Z",
              "description": "Diner",
              "from": "Wendy",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.ul
        className="list-group"
>
        <styled.li
                className="list-group-item active"
        >
                <span
                        className="col-xs-4"
                >
                        From
                </span>
                <span
                        className="col-xs-4"
                >
                        Description
                </span>
                <span
                        className="col-xs-4"
                >
                        Amount
                </span>
                <span
                        className="col-xs-4"
                >
                        Date
                </span>
        </styled.li>
        <styled.li
                className="list-group-item"
        >
                <span
                        className="col-xs-4"
                >
                        Mendy
                </span>
                <span
                        className="col-xs-4"
                >
                        Pizza
                </span>
                <span
                        className="col-xs-4"
                >
                        10.5
                </span>
                <span
                        className="col-xs-4"
                >
                        2017
                </span>
        </styled.li>
        <styled.li
                className="list-group-item"
        >
                <span
                        className="col-xs-4"
                >
                        Wendy
                </span>
                <span
                        className="col-xs-4"
                >
                        Diner
                </span>
                <span
                        className="col-xs-4"
                >
                        10.5
                </span>
                <span
                        className="col-xs-4"
                >
                        2016
                </span>
        </styled.li>
</styled.ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.li
            className="list-group-item active"
>
            <span
                        className="col-xs-4"
            >
                        From
            </span>
            <span
                        className="col-xs-4"
            >
                        Description
            </span>
            <span
                        className="col-xs-4"
            >
                        Amount
            </span>
            <span
                        className="col-xs-4"
            >
                        Date
            </span>
</styled.li>,
          Array [
            <styled.li
              className="list-group-item"
>
              <span
                            className="col-xs-4"
              >
                            Mendy
              </span>
              <span
                            className="col-xs-4"
              >
                            Pizza
              </span>
              <span
                            className="col-xs-4"
              >
                            10.5
              </span>
              <span
                            className="col-xs-4"
              >
                            2017
              </span>
</styled.li>,
            <styled.li
              className="list-group-item"
>
              <span
                            className="col-xs-4"
              >
                            Wendy
              </span>
              <span
                            className="col-xs-4"
              >
                            Diner
              </span>
              <span
                            className="col-xs-4"
              >
                            10.5
              </span>
              <span
                            className="col-xs-4"
              >
                            2016
              </span>
</styled.li>,
          ],
        ],
        "className": "list-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span
                className="col-xs-4"
>
                From
</span>,
              <span
                className="col-xs-4"
>
                Description
</span>,
              <span
                className="col-xs-4"
>
                Amount
</span>,
              <span
                className="col-xs-4"
>
                Date
</span>,
            ],
            "className": "list-group-item active",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "From",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Description",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "Description",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Amount",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "Amount",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "Date",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Pizza0",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span
                className="col-xs-4"
>
                Mendy
</span>,
              <span
                className="col-xs-4"
>
                Pizza
</span>,
              <span
                className="col-xs-4"
>
                10.5
</span>,
              <span
                className="col-xs-4"
>
                2017
</span>,
            ],
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mendy",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "Mendy",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pizza",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "Pizza",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 10.5,
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": 10.5,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2017,
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": 2017,
              "type": "span",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Diner1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span
                className="col-xs-4"
>
                Wendy
</span>,
              <span
                className="col-xs-4"
>
                Diner
</span>,
              <span
                className="col-xs-4"
>
                10.5
</span>,
              <span
                className="col-xs-4"
>
                2016
</span>,
            ],
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wendy",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "Wendy",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Diner",
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": "Diner",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 10.5,
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": 10.5,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2016,
                "className": "col-xs-4",
              },
              "ref": null,
              "rendered": 2016,
              "type": "span",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.ul
          className="list-group"
>
          <styled.li
                    className="list-group-item active"
          >
                    <span
                              className="col-xs-4"
                    >
                              From
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              Description
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              Amount
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              Date
                    </span>
          </styled.li>
          <styled.li
                    className="list-group-item"
          >
                    <span
                              className="col-xs-4"
                    >
                              Mendy
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              Pizza
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              10.5
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              2017
                    </span>
          </styled.li>
          <styled.li
                    className="list-group-item"
          >
                    <span
                              className="col-xs-4"
                    >
                              Wendy
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              Diner
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              10.5
                    </span>
                    <span
                              className="col-xs-4"
                    >
                              2016
                    </span>
          </styled.li>
</styled.ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.li
              className="list-group-item active"
>
              <span
                            className="col-xs-4"
              >
                            From
              </span>
              <span
                            className="col-xs-4"
              >
                            Description
              </span>
              <span
                            className="col-xs-4"
              >
                            Amount
              </span>
              <span
                            className="col-xs-4"
              >
                            Date
              </span>
</styled.li>,
            Array [
              <styled.li
                className="list-group-item"
>
                <span
                                className="col-xs-4"
                >
                                Mendy
                </span>
                <span
                                className="col-xs-4"
                >
                                Pizza
                </span>
                <span
                                className="col-xs-4"
                >
                                10.5
                </span>
                <span
                                className="col-xs-4"
                >
                                2017
                </span>
</styled.li>,
              <styled.li
                className="list-group-item"
>
                <span
                                className="col-xs-4"
                >
                                Wendy
                </span>
                <span
                                className="col-xs-4"
                >
                                Diner
                </span>
                <span
                                className="col-xs-4"
                >
                                10.5
                </span>
                <span
                                className="col-xs-4"
                >
                                2016
                </span>
</styled.li>,
            ],
          ],
          "className": "list-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span
                  className="col-xs-4"
>
                  From
</span>,
                <span
                  className="col-xs-4"
>
                  Description
</span>,
                <span
                  className="col-xs-4"
>
                  Amount
</span>,
                <span
                  className="col-xs-4"
>
                  Date
</span>,
              ],
              "className": "list-group-item active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "From",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "From",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "Description",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Amount",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "Amount",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Date",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "Date",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Pizza0",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span
                  className="col-xs-4"
>
                  Mendy
</span>,
                <span
                  className="col-xs-4"
>
                  Pizza
</span>,
                <span
                  className="col-xs-4"
>
                  10.5
</span>,
                <span
                  className="col-xs-4"
>
                  2017
</span>,
              ],
              "className": "list-group-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mendy",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "Mendy",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pizza",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "Pizza",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 10.5,
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": 10.5,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 2017,
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": 2017,
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Diner1",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span
                  className="col-xs-4"
>
                  Wendy
</span>,
                <span
                  className="col-xs-4"
>
                  Diner
</span>,
                <span
                  className="col-xs-4"
>
                  10.5
</span>,
                <span
                  className="col-xs-4"
>
                  2016
</span>,
              ],
              "className": "list-group-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Wendy",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "Wendy",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Diner",
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": "Diner",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 10.5,
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": 10.5,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 2016,
                  "className": "col-xs-4",
                },
                "ref": null,
                "rendered": 2016,
                "type": "span",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
